packages/server/chronos.ts

```diff
@@ -117,27 +117,34 @@ const chronos = () => {
   }
   Object.entries(jobs).forEach(([name, {onTick, cronTime}]) => {
     try {
-      CronJob.from({
+      const job = CronJob.from({
         start: true,
         // assume non-null & catch on fail
         cronTime: cronTime!,
         onTick
       })
+      runningJobs.push(job)
       Logger.log(`🌱 Chronos Job ${name}: STARTED`)
     } catch {
       Logger.log(`🌱 Chronos Job ${name}: SKIPPED`)
     }
   })
-
-  Logger.log(`\n🌾🌾🌾 Server ID: ${SERVER_ID}. Ready for Chronos           🌾🌾🌾`)
 }
 
-const runOnPrimary = async () => {
-  if (!__PRODUCTION__) return
+const runOnPrimary = () => {
+  if (!__PRODUCTION__) return () => {}
   const redis = new RedisInstance(`chronosLock_${SERVER_ID}`)
-  const primaryLock = await establishPrimaryServer(redis, 'chronos')
-  if (!primaryLock) return
-  chronos()
+  const leader = new LeaderElector(redis, 'chronos', 20_000)
+  leader.on('elected', () => {
+    Logger.log(`\n🌾🌾🌾 Server ID: ${SERVER_ID} Elected for Chronos           🌾🌾🌾`)
+    chronos()
+  })
+  leader.on('revoked', () => {
+    Logger.log(`\n🌾🌾🌾 Server ID: ${SERVER_ID} Lost Chronos           🌾🌾🌾`)
+    runningJobs.forEach((job) => job.stop())
+  })
+  leader.start()
+  return () => leader.stop()
```

Comment:
-1 this does not do `runningJobs.forEach((job) => job.stop())` like we do when it's revoked.